search-provider/eks-article-metadata-query-provider-dbus.xml

```diff
@@ -0,0 +1,99 @@
+  <interface name="com.endlessm.ArticleMetadata">
+    <!--
+        Query:
+        @Query: A dictionary describing the query to be made, with the following
+                optional parameters. Specifying any parameter that is not
+                a member of this list is an error. No new properties will
+                be added or removed from this interface during its lifetime.
+
+                "search-terms": A string (s) with the free-text search terms
+                                to use during the query. If the parameter
+                                is not specified, filtering will only
+                                be done against tags-match-any and tags-match-all.
+                "tags-match-any": A strv (as) with a list of tags that any
+                                  matching content objects must have at least
+                                  one of.
+                "tags-match-all": A strv (as) with a list of tags that any
+                                  matching content objects must have all of.
+                "limit": An integer indicating the number of results at most
+                         to return. If the parameter is not specifed there
+                         will be no limit.
+                "offset": An integer specifying the "offset" into the query
+                          results from which point ot start returning results.
+                          This also changes the upper bound if "limit" was
+                          specified. For instance, if "offset" was 1 and
+                          "limit" was 5, then the 2nd to 6th results will
+                          be returned, inclusive.
+
+       Run a query against the database for this app.
+
+       Returns a tuple of @Shards and @Results.
+       @Shards: An array of strings indicating the file-paths where
+                shard packfiles are kept on the filesystem which contain
+                the actual data for results in the query. A call to Query
+                might return multiple shards, so if the client wants to look
+                up further information for a particular content object, they
+                will need to check every shard to see if it has that content.
+       @Results: An array of dictionaries with metadata about each content
+                 object matched in the query. New propertis may be added
+                 to these dictionaries during the course of this interface's
+                 lifetime, so callers should explicitly check if the property
+                 is present on the returned dictionary for each piece of
+                 content before attempting to use it, and if not, substitute
+                 a sensible default.
+
+                 "title": The title of the content object.
+                 "content_type": The MIME type of the underlying content data.
+                 "thumbnail_uri": An EKN URI specifying the location in one
```

Comment:
Are both this and `ekn_id` "EKN URIs"? I thought this was a simple URI, and that `ekn_id` was an "EKN ID"; I'm not sure what an EKN URI is.

Reply:
Yes, they are both URIs (eg, `ekn:///some_hash`).

The naming convention here is a little unfortunate - I'm mirroring what eos-knowledge-lib uses here. On every ContentObjectModel we have a property called `ekn-id` which actually holds an `ekn:///` URI. Confusing, I know, but I decided to resolve it in favor of consistency.

Reply:
Not sure it's always the case in all current usage, but EKN ID is the preferred term and should include the `ekn:///`.

Reply:
OK, updated.